### PR TITLE
Fixed validation logic to allow paths without an extension

### PR DESCRIPTION
**Pull request by jayoshih**
Ricecooker will use a default_ext if the path doesn't end with one. However, validation requires it to have an extension. This PR fixes that

### CODE REVIEW

File: ricecooker/classes/files.py [hunk 1]
```diff
@@ -207,28 +207,43 @@ class ThumbnailFile(ThumbnailPresetMixin, DownloadFile):
     default_ext = file_formats.PNG
 
     def validate(self):
-        assert os.path.splitext(self.path)[1][1:] in [file_formats.JPG, file_formats.JPEG, file_formats.PNG], "Thumbnails must be in jpg, jpeg, or png format"
+        super(ThumbnailFile, self).validate()
+        if os.path.splitext(self.path)[1][1:] != "":
```
Comment (aronasorman) [inline ricecooker/classes/files.py]: I prefer this version:

```
basename, extension = os.path.splitext(...)
if extension:  # not empty
    ...
```

Comment (aronasorman) [inline ricecooker/classes/files.py]: Other lines on this file have this same pattern, maybe you can change it?

File: ricecooker/classes/files.py [hunk 2]
```diff
@@ -207,28 +207,43 @@ class ThumbnailFile(ThumbnailPresetMixin, DownloadFile):
...
+            assert self.path.endswith(file_formats.PDF), "Document files must be in pdf format"
 
 class HTMLZipFile(DownloadFile):
+    default_ext = file_formats.HTML5
+
     def get_preset(self):
         return self.preset or format_presets.HTML5_ZIP
 
     def validate(self):
```
Comment (aronasorman) [inline ricecooker/classes/files.py]: These `validate` functions also all have the same general logic. Can you find a way to just have them use the base class's `validate` command?

File: ricecooker/sample_program.py [hunk 3]
```diff
@@ -93,18 +103,18 @@ def guess_file_type(filepath, kind):
...
-                "path": "C:/users/jordan/Videos/testfolder/captions.vtt",
-                "language": languages.getlang('en').code,
-            }
-            ,
-            {
-                "path": "file:///C:/users/jordan/Videos/testfolder/captions.vtt",
-                "language": languages.getlang('es').code,
-            }
+            # {
+            #     "path": "C:/users/jordan/Videos/testfolder/captions.vtt",
```
Comment (aronasorman) [inline ricecooker/sample_program.py]: Delete commented out code

Comment (jayoshih) [inline ricecooker/sample_program.py]: Temporarily commented out since subtitles aren't allowed on the production server yet

**Pull request merged**